plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/BaseMongoConnectorTest.java

```diff
@@ -584,6 +586,71 @@ public void testAddColumnConcurrently()
         throw new SkipException("TODO");
     }
 
+    @Test
+    public void testNativeQuerySimple()
+    {
+        String id = (String) computeScalar("SELECT CAST(_id AS varchar) FROM region WHERE regionkey = 1");
+        @Language("JSON") String expected = "{\"_id\":{\"$oid\":\"" + id + "\"},\"comment\":\"hs use ironic, even requests. s\",\"name\":\"AMERICA\",\"regionkey\":1}";
+        assertThat(query("SELECT * FROM TABLE(mongodb.system.raw_query(database => 'tpch', collection => 'region', filter => '{ regionkey: 1 }'))"))
+                .matches("SELECT json '" + expected + "'");
+    }
+
+    @Test
+    public void testNativeQueryEmptyFilter()
+    {
+        // Empty filter '{}' returns all rows
+        assertThat(query("" +
+                "SELECT " +
+                "CAST(json_extract_scalar(result, '$.regionkey') AS bigint), " +
+                "CASt(json_extract_scalar(result, '$.name') AS varchar(25)), " +
+                "CAST(json_extract_scalar(result, '$.comment') AS varchar(152)) " +
+                "FROM TABLE(mongodb.system.raw_query(database => 'tpch', collection => 'region', filter => '{}'))"))
+                .matches("" +
+                        "SELECT " +
+                        "regionkey, " +
+                        "name, " +
+                        "comment " +
+                        "FROM region");
+    }
+
+    @Test
+    public void testNativeQuerySelectUnsupportedType()
+    {
+        // The table function allows reading unsupported types as json
```

Comment:
All the types are being read as JSON, but the PTF provides a way to inject custom filter. Do we need this restriction ?